mllib/src/main/scala/org/apache/spark/ml/feature/StringIndexer.scala

```diff
@@ -146,7 +146,7 @@ object StringIndexer extends DefaultParamsReadable[StringIndexer] {
  * This is a temporary fix for the case when target labels do not exist during prediction.
  */
 @Since("1.4.0")
-class StringIndexerModel (
+class StringIndexerModel private[ml] (
```

Comment:
I thought this was made public on purpose.  StringIndexerModel is one case where it makes sense for users to create it manually (if they already have a list of the possible Strings).